scripts/packages/VSCodeServer/src/display.jl

```diff
@@ -82,6 +87,41 @@ const DISPLAYABLE_MIMES = [
     "image/gif"
 ]
 
+"""
+    DIAGNOSTIC_MIME = "application/vnd.julia-vscode.diagnostics"
+
+User type needs to implement a `show` method that returns a named tuple or dictionary like the following
+```
+Base.show(io::IO, ::MIME"application/vnd.julia-vscode.diagnostics", t::YourType) = (
+    source = ""
+    items = [
+        (
+            msg = "foo",
+            path = "/some/absolute/path.jl",
+            line = 1 # 1 based
+            range = [[1, 2], [1, 4]] # or [[start_line, start_char], [end_line, end_char]],
+            severity = 1, # optional; 0: Error, 1: Warning, 2: Information, 3: Hint
+            relatedInformation = [
+                (
+                    msg = "foobar",
+                    path = "/some/other/absolute/path.jl",
+                    line = 1,
+                    range = [[1, 2], [1, 4]] # or [[start_line, start_char], [end_line, end_char]],
+                )
+            ] # optional
```

Comment:
I think this isn't actually optional ?
https://github.com/julia-vscode/julia-vscode/blob/fa7ab65ae1c94ef50334cd3cf1bcd9983ff2bc2f/src/interactive/repl.ts#L424

Reply:
Ah. It should be though :P